### ISSUE TITLE
Aborting the http request does not abort async resolvers

### ISSUE THREAD

**Issue by jarlef** [ChilliCream/graphql-platform#7464]
### Product

Hot Chocolate

### Version

13.9.12

### Link to minimal reproduction

https://github.com/jarlef/HC-Bug-CancellationToken

### Steps to reproduce

- Run server and query the foo field resolver in BananaCakePop
- Click "Cancel" immediately
- Observe HC has returned ClientAbortError but the resolver is still being executed without raising `OperationCanceledException`  

### What is expected?

HC should raise `OperationCanceledException`on the CancellationToken provided to the async resolvers

### What is actually happening?

The client receives an instant confirmation that the request has been aborted but HC server is still processing 

### Additional context

Have also added a asp .net minimal api get endpoint which performs the same logic, but it manages to raise the OperationCanceledException and abort the request properly.

Not this also affects older version of HC down to at least 13.5.0 (or even further) and also 14.0.0-rc.1

This is real concern for APIs that are invoking slower upstream apis that the resolvers are not terminated and thereby performing heavy computation even if the client has determined the computation is no longer needed. Essentially clients could requests could create a DDoS situation.

**Comment by michaelstaib** [comment]
This is not a Hot Chocolate bug but a BCP issue ... @rstaib it seams that the request is not correctly aborted in BCP. I checked with postman and aborting the request in postman leads to the correct cancellation.

**Comment by jarlef** [comment]
We first notice the same issue using react apollo where we call the abort on the query request. The requests turns up as cancelled in chrome but the server still processes the request. Will check this further if we can reproduce this.

**Comment by michaelstaib** [comment]
So we tested it now intensely and it works like with BCP, react and postman. We do not see any issue. So we cannot reproduce the issue you're having.

Sometimes  this is cause by an invalid request middleware or by http request interceptor that does not pass the original cancellation token on.

I am closing this issue as this seem a user problem. If you however produce a repro that is verifiable you can file a new issue.

**Comment by jarlef** [comment]
@michaelstaib  Turns out my repro was inaccurate. Seems like aborting requests in banana cake pop (or in general) in Brave does not trigger  asp .net to raising the OperationCanceledException. That seems to be a browser issue (i'm going to switch browser from now on)

However. I found out that async resolvers aborts when the request is aborted in chrome. But if the async resolver is using dataloader, then the dataloader is not aborted.  I have updated with a repro with an example using dataloaders. I have attached a http file with the different request (with and without data loaders)
